sklearn/ensemble/_hist_gradient_boosting/splitting.pyx

```diff
@@ -462,7 +463,7 @@ cdef class Splitter:
                 # won't get any better (hessians are > 0 since loss is convex)
                 break
 
-            gain = _split_gain(sum_gradient_left, sum_hessian_left,
+            gain += _split_gain(sum_gradient_left, sum_hessian_left,
```

Comment:
you're incrementing `gain` over and over instead of incrementing the value from `-negative_loss(sum_gradients, sum_hessians, self.l2_regularization)`.